src/Digdir.Library.Utils.AspNet/HealthCheckExtensions.cs

```diff
@@ -0,0 +1,78 @@
+using Digdir.Library.Utils.AspNet.HealthChecks;
+using HealthChecks.UI.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics.HealthChecks;
+
+namespace Digdir.Library.Utils.AspNet;
+
+public static class HealthCheckExtensions
+{
+    public class AspNetHealthChecksOptions
+    {
+        public string WellKnownEndpointsConfigurationSectionPath { get; set; } = string.Empty;
+    }
+
+    public class JwtBearerTokenSchema
+    {
+        public string Name { get; set; } = string.Empty;
+        public string WellKnown { get; set; } = string.Empty;
+    }
+
+    public static IServiceCollection AddAspNetHealthChecks(this IServiceCollection services, IConfiguration configuration, Action<AspNetHealthChecksOptions> configure)
+    {
+        var options = new AspNetHealthChecksOptions();
+        configure(options);
+
+        var wellKnownSchemas = configuration
+            .GetSection(options.WellKnownEndpointsConfigurationSectionPath)
+            .Get<List<JwtBearerTokenSchema>>();
+
+        var healthChecks = services.AddHealthChecks();
+
+        healthChecks.AddCheck("self", () => HealthCheckResult.Healthy(), tags: ["self"]);
+
+        var wellKnownEndpoints = wellKnownSchemas?.Select(schema => schema.WellKnown).ToList() ?? new List<string>();
+
+        if (wellKnownEndpoints.Count > 0)
+        {
+            services.Configure<EndpointsHealthCheckOptions>(opts =>
+            {
+                opts.Endpoints = wellKnownEndpoints;
+            });
+
+            healthChecks.AddCheck<EndpointsHealthCheck>(
+                "Endpoints",
+                failureStatus: HealthStatus.Unhealthy,
+                tags: ["dependencies"]);
+        }
+
+        return services;
+    }
+
+    public static void MapAspNetHealthChecks(this WebApplication app)
+    {
+        app.MapHealthChecks("/startup", new HealthCheckOptions
+        {
+            Predicate = check => check.Tags.Contains("dependencies"),
```

Comment:
Hadde vært fint om du la til disse taggene som konstanter. 🙏

Reply:
Trenger ikke gjøre det med endepunktene

Reply:
Yes. Brukes både i infrastructure og webapi, så kanskje skille det ut i et lib? Evt. deklarere konstanter i begge prosjektene da.

Reply:
Hmm, det er jo et godt poeng. Dersom vi eksponerer det i en public klasse her er det kun tilgjengelig for WebApi og ikke infrastruktur. Det "riktigste" hadde vært å lage en `Digdir.Library.Utils.Abstractions` pakke som WebApi hadde brukt implisitt gjennom `Digdir.Library.Utils.AspNet` avhengighet og Infrastruktur hadde brukt eksplistt. Men det er kanskje overkill for _en_ fil? 🤔 Et annet alternativ er å gjøre et kompromiss. public static class i `Digdir.Library.Utils.AspNet` som WebApi bruker, og magic strings i Infrastruktur. Jeg lar det være opp til deg 😬